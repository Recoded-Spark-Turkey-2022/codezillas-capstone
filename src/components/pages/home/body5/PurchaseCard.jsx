import React from 'react';
import {Link } from 'react-router-dom';

function PurchaseCard(props) {
  const { title, dollar, text, button, classList,url } = props;
  return (
    <div
      className={
        !classList
          ? 'border border-solid w-60 xl:w-80 2xl:w-96 border-gray-200 rounded-lg shadow-xl'
          : classList
      }
    >
      <div className="flex flex-col align-center m-6">
        <h1 className="text-3xl font-semibold pt-12  flex flex-col items-center justify-center 2xl:text-5xl">
          {title}
        </h1>
        <h2 className="text-gray-400 text-xl pt-3 font-medium flex flex-col items-center justify-center 2xl:text-3xl 2xl:py-8">
          {dollar}
        </h2>
        <h2 className="text-base pt-2 font-normal 2xl:text-2xl pb-6">{text}</h2>

        <div className="flex flex-col items-center justify-center pb-10 2xl:text-3xl 2xl:pb-12">
        <Link  to={url}>
          <input
            type="button"
            value={button}
            className="py-2 px-7 hover:bg-cyan-200 bg-cyan-400 font-medium rounded"
          />
          </Link>
          </div>
        
        </div>
      </div>
    
  );
}

export default PurchaseCard;
