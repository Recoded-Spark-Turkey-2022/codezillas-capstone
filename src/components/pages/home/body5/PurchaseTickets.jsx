import React from 'react';
import {Link } from 'react-router-dom';
import PurchaseCard from './PurchaseCard';
import Button from '../../../shared/button/Button';
/* Purchase button should take us to the Select Card and Learn More button shoul navigate to the Why Work With Healing? */
function PurchaseTickets(props) {
  const {auth} = props
  let url;
  if (auth) {
    url = 'select-card'
  }else {
    url = 'login'
  }

  return (
    <div className="">
      <div className="">
        <div className="flex flex-col justify-between sm:items-center lg:items-start lg:px-24 xl:px-32 2xl:px-48">
          <p className="text-4xl mt-12 mb-6 text-center lg:text-start 2xl:text-6xl font-bold ">
            PURCHASE TICKETS
          </p>
          <p className="text-l font-normal mb-16 text-center lg:text-start 2xl:text-3xl ">
            PURCHASE TICKETS THAT CAN BE USED TO BOOK APPOINTMENTS!
          </p>
        </div>

        <div className="flex flex-col gap-4 2xl:gap-20 items-center ">
          <div className="flex flex-wrap justify-center gap-4 lg:gap-14 xl:gap-32 2xl:gap-20">
            <PurchaseCard title="5 TICKETS" dollar="10$" button="PURCHASE" url={url} />

            <PurchaseCard title="25 TICKETS" dollar="40$" button="PURCHASE" url={url} />

            <PurchaseCard title="50 TICKETS" dollar="70$" button="PURCHASE" url={url} />
          </div>

        <div className="flex flex-col text-center mb-10 border border-solid w-5/6 border-gray-200 rounded-lg shadow-xl">
        <h1 className="text-3xl font-semibold pt-12  flex flex-col items-center justify-center 2xl:text-5xl">
        ARE YOU A COUNSELOR?
        </h1>
        <h2 className="text-base pt-2 font-normal 2xl:text-2xl pb-6"> 
        interested in joining our mental health platform? you decide your schedule and how much you want to work, we&apos;ll take care of the client referrals and billing details!
        </h2>

        <div className="flex flex-col items-center justify-center pb-10 2xl:text-3xl 2xl:pb-12">
        <Link  to="/work-with-healing">
        <Button
              type="button"
              name="LEARN MORE"
              classList="py-2 px-7 hover:bg-cyan-200 bg-cyan-400 font-medium rounded"
            />
          </Link>
          </div>
        
        </div>


        </div>
      </div>
    </div>
  );
}


export default PurchaseTickets;
