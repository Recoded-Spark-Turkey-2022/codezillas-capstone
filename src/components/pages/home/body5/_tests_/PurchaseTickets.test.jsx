import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import PurchaseTickets from "../PurchaseTickets";
import PurchaseCard from "../PurchaseCard";


test('PurchaseTickets component renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
    <PurchaseCard />
    <PurchaseTickets />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
