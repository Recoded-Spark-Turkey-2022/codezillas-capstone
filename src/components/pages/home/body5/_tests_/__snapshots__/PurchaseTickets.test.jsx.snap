// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PurchaseTickets component renders correctly 1`] = `
Array [
  <div
    className="border border-solid w-60 xl:w-80 2xl:w-96 border-gray-200 rounded-lg shadow-xl"
  >
    <div
      className="flex flex-col align-center m-6"
    >
      <h1
        className="text-3xl font-semibold pt-12  flex flex-col items-center justify-center 2xl:text-5xl"
      />
      <h2
        className="text-gray-400 text-xl pt-3 font-medium flex flex-col items-center justify-center 2xl:text-3xl 2xl:py-8"
      />
      <h2
        className="text-base pt-2 font-normal 2xl:text-2xl pb-6"
      />
      <div
        className="flex flex-col items-center justify-center pb-10 2xl:text-3xl 2xl:pb-12"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <input
            className="py-2 px-7 hover:bg-cyan-200 bg-cyan-400 font-medium rounded"
            type="button"
          />
        </a>
      </div>
    </div>
  </div>,
  <div
    className=""
  >
    <div
      className=""
    >
      <div
        className="flex flex-col justify-between sm:items-center lg:items-start lg:px-24 xl:px-32 2xl:px-48"
      >
        <p
          className="text-4xl mt-12 mb-6 text-center lg:text-start 2xl:text-6xl font-bold "
        >
          PURCHASE TICKETS
        </p>
        <p
          className="text-l font-normal mb-16 text-center lg:text-start 2xl:text-3xl "
        >
          PURCHASE TICKETS THAT CAN BE USED TO BOOK APPOINTMENTS!
        </p>
      </div>
      <div
        className="flex flex-col gap-4 2xl:gap-20 items-center "
      >
        <div
          className="flex flex-wrap justify-center gap-4 lg:gap-14 xl:gap-32 2xl:gap-20"
        >
          <div
            className="border border-solid w-60 xl:w-80 2xl:w-96 border-gray-200 rounded-lg shadow-xl"
          >
            <div
              className="flex flex-col align-center m-6"
            >
              <h1
                className="text-3xl font-semibold pt-12  flex flex-col items-center justify-center 2xl:text-5xl"
              >
                5 TICKETS
              </h1>
              <h2
                className="text-gray-400 text-xl pt-3 font-medium flex flex-col items-center justify-center 2xl:text-3xl 2xl:py-8"
              >
                10$
              </h2>
              <h2
                className="text-base pt-2 font-normal 2xl:text-2xl pb-6"
              />
              <div
                className="flex flex-col items-center justify-center pb-10 2xl:text-3xl 2xl:pb-12"
              >
                <a
                  href="/login"
                  onClick={[Function]}
                >
                  <input
                    className="py-2 px-7 hover:bg-cyan-200 bg-cyan-400 font-medium rounded"
                    type="button"
                    value="PURCHASE"
                  />
                </a>
              </div>
            </div>
          </div>
          <div
            className="border border-solid w-60 xl:w-80 2xl:w-96 border-gray-200 rounded-lg shadow-xl"
          >
            <div
              className="flex flex-col align-center m-6"
            >
              <h1
                className="text-3xl font-semibold pt-12  flex flex-col items-center justify-center 2xl:text-5xl"
              >
                25 TICKETS
              </h1>
              <h2
                className="text-gray-400 text-xl pt-3 font-medium flex flex-col items-center justify-center 2xl:text-3xl 2xl:py-8"
              >
                40$
              </h2>
              <h2
                className="text-base pt-2 font-normal 2xl:text-2xl pb-6"
              />
              <div
                className="flex flex-col items-center justify-center pb-10 2xl:text-3xl 2xl:pb-12"
              >
                <a
                  href="/login"
                  onClick={[Function]}
                >
                  <input
                    className="py-2 px-7 hover:bg-cyan-200 bg-cyan-400 font-medium rounded"
                    type="button"
                    value="PURCHASE"
                  />
                </a>
              </div>
            </div>
          </div>
          <div
            className="border border-solid w-60 xl:w-80 2xl:w-96 border-gray-200 rounded-lg shadow-xl"
          >
            <div
              className="flex flex-col align-center m-6"
            >
              <h1
                className="text-3xl font-semibold pt-12  flex flex-col items-center justify-center 2xl:text-5xl"
              >
                50 TICKETS
              </h1>
              <h2
                className="text-gray-400 text-xl pt-3 font-medium flex flex-col items-center justify-center 2xl:text-3xl 2xl:py-8"
              >
                70$
              </h2>
              <h2
                className="text-base pt-2 font-normal 2xl:text-2xl pb-6"
              />
              <div
                className="flex flex-col items-center justify-center pb-10 2xl:text-3xl 2xl:pb-12"
              >
                <a
                  href="/login"
                  onClick={[Function]}
                >
                  <input
                    className="py-2 px-7 hover:bg-cyan-200 bg-cyan-400 font-medium rounded"
                    type="button"
                    value="PURCHASE"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="flex flex-col text-center mb-10 border border-solid w-5/6 border-gray-200 rounded-lg shadow-xl"
        >
          <h1
            className="text-3xl font-semibold pt-12  flex flex-col items-center justify-center 2xl:text-5xl"
          >
            ARE YOU A COUNSELOR?
          </h1>
          <h2
            className="text-base pt-2 font-normal 2xl:text-2xl pb-6"
          >
            interested in joining our mental health platform? you decide your schedule and how much you want to work, we'll take care of the client referrals and billing details!
          </h2>
          <div
            className="flex flex-col items-center justify-center pb-10 2xl:text-3xl 2xl:pb-12"
          >
            <a
              href="/work-with-healing"
              onClick={[Function]}
            >
              <input
                className="py-2 px-7 hover:bg-cyan-200 bg-cyan-400 font-medium rounded"
                onClick={[Function]}
                type="button"
                value="LEARN MORE"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
